import_craft.py

```diff
@@ -0,0 +1,291 @@
+                if value[0] == 'part':
+                    self.nb_total_parts += 1
+                    part_name = value[1].rsplit('_', 1)[0].replace('_', " ").replace(".", "_")
+                    if part_name in parts_files:
+                            part['name'] = part_name
+                            part['key'] = value[1].rsplit('_', 1)[1]
+                            if part_name not in self.prefabs:
+                                self.prefabs.append(part_name)
+                    else:
+                        print('Warning: part {} was avoided since the mu is missing'.format(value[1]))
+                        return None
+
+                elif value[0] in ['pos', 'mir']:
+                    part[value[0]] = read_vector(value[1])
+                elif value[0] in ['rot', 'attRot']:
+                    part[value[0]] = read_quaternion(value[1])
+                else:
+                    part[value[0]] = value[1]
+
+        return part
+
+    def read_craft_file(self, filepath, parts_files):
+        ''' Read craft file nodes and returns parts list'''
+        with open(filepath, 'r') as craft_file:
+            craft_data = craft_file.read()
+
+        cfgnode = ConfigNode.load(craft_data)
+        # The first value must be the ship name
+        self.ship_name = cfgnode.values[0][1]
+
+        craft_parts = []
+        for node in cfgnode.nodes:
+            part = self.read_craft_node(node, parts_files)
+            if part is not None:
+                craft_parts.append(part)
+
+        return craft_parts
+
+    def set_craft_data(self, parts, partsdict):
+        ''' Generate parts blender object from craft parts list '''
+        # Contains the generated prefab's blender objects
+        # to be deleted at the end
+        originals = []
+        print_blender_data_stats('originals')
+        for obj in bpy.context.scene.objects:
+            if obj.parent is None:
+                originals.append(obj)
+
+        # create root node with craft(ship) name
+        main = bpy.data.objects.new(self.ship_name, object_data=None)
+        bpy.context.scene.objects.link(main)
+        unselect_all_objects()
+
+        self.generate_parts(parts, main, partsdict)
+        remove_object_list(originals)
+
+        return {'FINISHED'}
+
+
+def check_parts_in_directory(directory):
+    ''' Scan the directory for parts (pairs of .cfg and .mu files) '''
+    parts_files = dict()
+    for root, dirs, files in os.walk(directory):
+        for name in files:
+            if name.split('.')[-1] == 'cfg':
+                cfg_file_path = os.path.join(root, name)
+                with open(cfg_file_path, 'r') as partcfg:
+                    try:
+                        partcfg_data = partcfg.read()
+                        cfgnode = ConfigNode.load(partcfg_data)
+                    except UnicodeDecodeError:
+                        print('Bad encoding found while loading part.cfg')
+                        break
```

Comment:
why break? we could maybe try to process the other parts? especially if we just break we would still process the files previously found